Add matchEntityType(String) method to EntityType class. Adds BUKKIT-5777

caseif opened the pull request:
##### The Issue:

The EntityType enum, while mostly identical to the Material enum, lacks a method corresponding to Material#matchMaterial(String).
##### Justification:

In the interest of consistency and ease of use, a corresponding method is beneficial to the API.
##### PR Breakdown:

Creates a method in EntityType called matchEntityType(String), which formats the argument appropriately and matches it to a corresponding value of the enum. Returns null if none is found.
##### Testing Results and Materials:

See EntityTypeTest.java.
##### JIRA Ticket:

Bukkit-5777 - https://bukkit.atlassian.net/browse/BUKKIT-5777?filter=10004

caseif committed:
Add matchEntityType(String) method to EntityType class. Adds BUKKIT-5777

caseif committed:
Correct documentation

ST-DDT commented:
I guess testing results and materials would be a JUnit test or a testplugin with sources.

caseif committed:
Fix matcher method not working with raw enum name
Write unit tests for added methods
Correct typo in ocelot name in EntityType.java (non-breaking)

caseif committed:
Apparently that was NOT a typo

caseif commented:
@ST-DDT I've added some unit tests pertaining to the method.

bendem commented on src/main/java/org/bukkit/entity/EntityType.java:
```diff
@@ -248,6 +252,42 @@ public static EntityType fromId(int id) {
+            filtered = filtered.replaceAll("\\s+", "_").replaceAll("\\W", "");
+            result = fromName(filtered);
+            if (result == null)
+            	try {
```
There are tabs here...

caseif replied:
@bendem My mistake; Eclipse adds them automatically. I'll fix it momentarily.

bendem commented on src/test/java/org/bukkit/EntityTypeTest.java:
```diff
@@ -0,0 +1,89 @@
+    @Test
+    public void getByName() {
+        for (EntityType entityType : EntityType.values()) {
+        	if (entityType.getName() != null)
```
There's a tab here

bendem commented on src/test/java/org/bukkit/EntityTypeTest.java:
```diff
@@ -0,0 +1,89 @@
+    @Test
+    public void matchEntityTypeByName() {
+        for (EntityType entityType : EntityType.values()) {
+        	if (entityType.getName() != null)
```
There is a tab here

bendem commented on src/main/java/org/bukkit/entity/EntityType.java:
```diff
@@ -248,6 +252,42 @@ public static EntityType fromId(int id) {
+            	try {
+            	    result = EntityType.valueOf(filtered.toUpperCase());
+            	}
+                catch (IllegalArgumentException ex){} // okay to swallow it; we'll just return null
```
missing a space before `{}` and I think catch should be on the same line as the brace

caseif replied:
@bendem I believe that's a matter of style, though you're correct about the missing space.

bendem replied:
I'm not referring about my personal preferences but about the rest of the Bukkit codebase ;)

bendem commented on src/main/java/org/bukkit/entity/EntityType.java:
```diff
@@ -248,6 +252,42 @@ public static EntityType fromId(int id) {
         }
         return ID_MAP.get((short) id);
     }
+    
```
Trailing spaces here

bendem commented:
There are a few formatting problems, also I think you should use braces even for one statement if code. :-)

Pokechu22 commented:
FYI, eclipse has a feature to automatically format this kind of stuff under `Window->Preferences`, and then `Java->Code Style->Clean Up`.  You can run it by right-clicking on the file or in the editor, and then selecting `Source->Clean Up...`.  Might be helpful.
